app/controllers/pet_applications_controller.rb

```diff
@@ -7,9 +7,19 @@ def index
   def update
     pet = Pet.find(params[:id])
     application = Application.find(params[:app_id])
-    pet.update_adoption_status(pet, params)
-    flash[:notice] = "#{pet.name} is on hold for #{application.name}."
+    if pet.adoption_status == "Pending"
+      flash[:notice] = "No more applications can be approved for #{pet.name} at this time"
+    else
+      pet.update_adoption_status("Pending", true, params)
+      flash[:notice] = "#{pet.name} is on hold for #{application.name}."
+    end
     redirect_to "/pets/#{pet.id}"
   end
 
+  def destroy
+    pet = Pet.find(params[:id])
+    application = Application.find(params[:app_id])
+    pet.update_adoption_status("Adoptable", false, params)
```

Comment:
If we have time, it might be in our interest to remove the data type 'string' all together, and just pass it false